languages/js/src/Query.ts

```diff
@@ -359,17 +363,44 @@ export class Query {
             }
             const { message } = event.data as Debug;
             if (message) console.log(message);
-            createInterface({
-              input: process.stdin,
-              output: process.stdout,
-              prompt: 'debug> ',
-              tabSize: 4,
-            }).on('line', (line: string) => {
-              const trimmed = line.trim().replace(/;+$/, '');
-              const command = this.#host.toPolar(trimmed);
-              this.#ffiQuery.debugCommand(JSON.stringify(command));
-              this.processMessages();
+
+            if (typeof rl === 'undefined') {
+              // @ts-ignore
+              rl = global.oso_repl_rl;
+              if (rl) {
+                const exitListeners = rl.listeners('exit') as (() => void)[];
```

Comment:
Messy stuff to swap event handlers from the REPL ones to debugger ones and back. Kind of like what we do for the node repl.

Reply:
Took me a second to understand what this is doing -- can you make this a comment? And maybe flesh it out a bit, like `If we hit a debug event during a REPL session, we need to re-use the existing readline interface to prevent double inputs. This code updates the readline interface to use a debug prompt, until the debug prompt exits.`